### PR TITLE
Delete log records when a model is removed

### PR DESCRIPTION
**Pull request by babbageclunk**
It became apparent that they were being left behind after migration, and it turns out also after model destruction (although they'd only be accessible through the Mongo shell). 

QA steps:
* bootstrap a controller
* add a model and deploy a charm like ubuntu to it (to generate model logs)
* check how many log messages there are for the model
* run `juju show-model` and note the UUID
* destroy the model
* confirm there are no log records for the model left in the MongoDB shell. (See https://github.com/juju/juju/wiki/Login-into-MongoDB to connect to the shell.)
```
use logs
db.logs.find({"e": "<model-uuid>"}).count()
```

### CODE REVIEW

**Comment by babbageclunk** [comment]
!!build!!

File: state/state_test.go [hunk 1]
```diff
@@ -2631,13 +2651,46 @@ func (s *StateSuite) TestRemoveExportingModelDocsExporting(c *gc.C) {
...
 	err = st.RemoveExportingModelDocs()
 	c.Assert(err, jc.ErrorIsNil)
 
 	// test that we can not find the user:envName unique index
 	s.checkUserModelNameExists(c, checkUserModelNameArgs{st: st, id: userModelKey, exists: false})
 	s.AssertModelDeleted(c, st)
 	c.Assert(state.HostedModelCount(c, s.State), gc.Equals, 0)
+
+	// These are still there.
+	assertLogCount(c, s.State, 5)
```
Comment (howbazaar) [inline state/state_test.go]: Do we really need to test this three times each for removed and not?

Worthwhile making separate tests to show log removal or not?

Comment (babbageclunk) [inline state/state_test.go]: Gah, I had them as three separate tests first and then decided that was silly for reasons that I can't really articulate. I'll split them out again. I don't mind testing it just for one of them, although it seemed a bit arbitrary which one.

**Comment by babbageclunk** [comment]
$$merge$$

**Comment by babbageclunk** [comment]
Ha ha, I still had the split-out tests in my undo history.

**Pull request merged**